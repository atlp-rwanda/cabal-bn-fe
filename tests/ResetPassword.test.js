import * as React from 'react';
import '@testing-library/jest-dom';
import {
  render,
  screen,
  waitFor,
  fireEvent,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import ResetPassword from '../src/view/ResetPassword';
import store from '../src/redux/store';
import { Reset } from '../src/redux/actions/ResetPassword.action';

describe('RESET PASSWORD TEST', () => {
  it('should renders Reset Password component', () => {
    const element = renderer
      .create(
        <Provider store={store}>
          <Router>
            <ResetPassword />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });
  it('should testing invalid empty password', () => {
    render(
      <Provider store={store}>
        <Router>
          <ResetPassword />
        </Router>
      </Provider>,
    );
    const email = screen.getByLabelText('password');
    fireEvent.change(email, { target: { value: '' } });
    const button = screen.getByRole('button', { name: 'Change Password' });
    fireEvent.click(button);
  });
  it('should testing valid password and comfirm password', () => {
    render(
      <Provider store={store}>
        <Router>
          <ResetPassword />
        </Router>
      </Provider>,
    );
    const password = screen.getByLabelText('password');
    fireEvent.change(password, { target: { value: 'Password123' } });
    const Confirmpassword = screen.getByLabelText('Confirm password');
    fireEvent.change(Confirmpassword, { target: { value: 'Password123' } });
    const button = screen.getByRole('button', { name: 'Change Password' });
    fireEvent.click(button);
    expect(Reset(password));
  });
  it('should testing Password and confirm password does not match', async () => {
    render(
      <Provider store={store}>
        <Router>
          <ResetPassword />
        </Router>
      </Provider>,
    );
    const password = screen.getByLabelText('password');
    fireEvent.change(password, { target: { value: 'Password12355' } });
    const Confirmpassword = screen.getByLabelText('Confirm password');
    fireEvent.change(Confirmpassword, { target: { value: 'Password123' } });
    const button = screen.getByRole('button', { name: 'Change Password' });
    fireEvent.click(button);
    waitFor(async () => {
      expect(
        'Password and confirm password does not match',
      ).toBeInTheDocument();
    });
  });

  test('show password', async () => {
    render(
      <Provider store={store}>
        <Router>
          <ResetPassword />
        </Router>
      </Provider>,
    );
    const visibilityIcon = screen.getByTestId('visibility-button');
    const password = screen.getByLabelText('password');
    fireEvent.change(password, { target: { value: 'Wordpass123' } });
    userEvent.click(visibilityIcon);
    await waitFor(() => {
      expect(password).toHaveAttribute('type', 'text');
    });
  });

  test.skip('show comfirm password', async () => {
    render(
      <Provider store={store}>
        <Router>
          <ResetPassword />
        </Router>
      </Provider>,
    );
    const visibilityIcon = screen.getByTestId('visibility-button1');
    const password = screen.getByLabelText('Confirm password');
    fireEvent.change(password, { target: { value: 'Wordpass123' } });
    userEvent.click(visibilityIcon);
    await waitFor(() => {
      expect(password).toHaveAttribute('type', 'text');
    });
  });
});
