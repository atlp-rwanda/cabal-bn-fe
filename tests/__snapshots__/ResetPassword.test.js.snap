// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESET PASSWORD TEST should renders Reset Password component 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-131qkm5-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-j8xoiz-MuiToolbar-root"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="#"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <img
          alt="Logo"
          src="Logo.svg"
          style={
            Object {
              "height": 50,
              "width": 100,
            }
          }
        />
      </a>
      <div
        className="MuiBox-root css-9np8v5"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-hfh7pe-MuiTypography-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-pq7q13-MuiTypography-root-MuiLink-root"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Home
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-hfh7pe-MuiTypography-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-pq7q13-MuiTypography-root-MuiLink-root"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            About Us
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-hfh7pe-MuiTypography-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-pq7q13-MuiTypography-root-MuiLink-root"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Accommodation
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-hfh7pe-MuiTypography-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-pq7q13-MuiTypography-root-MuiLink-root"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Sign In
          </a>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-hfh7pe-MuiTypography-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-pq7q13-MuiTypography-root-MuiLink-root"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Sign Up
          </a>
        </p>
      </div>
      <div
        className="MuiBox-root css-1be6mpr"
        onClick={[Function]}
      >
        <img
          alt="menu bar"
          src="MenuBar.svg"
        />
      </div>
    </div>
  </header>,
  <div
    className="MuiBox-root css-1p411wg"
  >
    <div
      className="MuiBox-root css-rntpfg"
    >
      <img
        alt="login"
        className="css-7vartw"
        src="Group 47.svg"
      />
    </div>
    <div
      className="MuiBox-root css-reitts"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-egmpiu-MuiTypography-root"
      >
        Reset Password
      </h4>
      <div
        className="css-s3cggk-MuiStack-root"
      />
      <div
        className="MuiFormControl-root MuiTextField-root css-1x1ecdk-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          password
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              data-testid="visibility-button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-zs3ucl-MuiSvgIcon-root"
                data-testid="VisibilityOffIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root css-1x1ecdk-MuiFormControl-root-MuiTextField-root"
        data-testid="Confirmpassword"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r1:"
          id=":r1:-label"
        >
          Confirm password
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r1:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              data-testid="visibility-button1"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-zs3ucl-MuiSvgIcon-root"
                data-testid="VisibilityOffIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Confirm password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-jzmx1y-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Change Password
      </button>
      <div
        className="MuiBox-root css-vtpeil"
      />
    </div>
  </div>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-1mb05um-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-9x7x8m-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item css-1mha57u-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            className="css-le9e34-MuiStack-root"
          >
            <img
              alt="logo"
              height="62px"
              src="Logo.svg"
              width="130.2px"
            />
            <p
              className="MuiTypography-root MuiTypography-body1 css-9wmnc6-MuiTypography-root"
            >
              Land behold it created good saw after she&aposd Our set living. Signs midst dominion creepeth morning laboris nisi ufsit aliquip.
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            className="css-1f85j5c-MuiStack-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-s1fqaa-MuiTypography-root"
            >
              Navigation
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              About Us
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              Accommodation
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              SignIn
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              Sign Up
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            className="css-1f85j5c-MuiStack-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-s1fqaa-MuiTypography-root"
            >
              Services
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              Hotel Booking
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              Trip Booking
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
            >
              Destination Booking
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-s1fqaa-MuiTypography-root"
          >
            Contact
          </p>
          <div
            className="css-5jfrzp-MuiStack-root"
          >
            <img
              alt="facebook icon"
              className="css-4b494m"
              src="facebook-icon.svg"
            />
            <img
              alt="twitter icon"
              className="css-4b494m"
              src="twitter-icon.svg"
            />
            <img
              alt="gmail icon"
              className="css-4b494m"
              src="gmail-icon.svg"
            />
            <img
              alt="linkedin icon"
              className="css-4b494m"
              src="linkedin-icon.svg"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1235pdo-MuiTypography-root"
        >
          Copyright ©2022 All rights reserved | This template is made by cabal
        </p>
      </div>
    </div>
  </div>,
]
`;
